Algorithm/QCAlgorithm.Universe.cs

```diff
@@ -429,19 +434,26 @@ public void AddUniverse(SecurityType securityType, string name, Resolution resol
             var exchangeTimeZone = marketHoursDbEntry.ExchangeHours.TimeZone;
             var symbol = QuantConnect.Symbol.Create(name, securityType, market);
             var config = new SubscriptionDataConfig(typeof(CoarseFundamental), symbol, resolution, dataTimeZone, exchangeTimeZone, false, false, true, isFilteredSubscription: false);
-            AddUniverse(new UserDefinedUniverse(config, universeSettings, SecurityInitializer, resolution.ToTimeSpan(), selector));
+            AddUniverse(new UserDefinedUniverse(config, universeSettings, resolution.ToTimeSpan(), selector));
         }
 
         /// <summary>
-        /// Adds the security to the user defined universe for the specified
+        /// Adds the security to the user defined universe
         /// </summary>
-        private void AddToUserDefinedUniverse(Security security)
+        /// <param name="security">The security to add</param>
+        /// <param name="resolution">The <see cref="Resolution"/> requested, Tick, Second, Minute, Hour, or Daily</param>
+        /// <param name="leverage">The requested leverage for this security.</param>
+        /// <param name="market">The market of the security</param>
+        /// <param name="isFillDataForward">If true, returns the last available data even if none in that TimeSlice.</param>
+        /// <param name="isExtendedMarketHours">ExtendedMarketHours send in data from 4am - 8pm, not used for FOREX</param>
+        /// <param name="isInternalFeed">Setting this flag to true will prevent the data from being sent into the algorithm's OnData methods</param>
+        private void AddToUserDefinedUniverse(Security security, Resolution resolution, decimal leverage, string market, bool isFillDataForward, bool isExtendedMarketHours, bool isInternalFeed)
         {
             // if we are adding a non-internal security which already has an internal feed, we remove it first
             Security existingSecurity;
             if (Securities.TryGetValue(security.Symbol, out existingSecurity))
             {
-                if (!security.IsInternalFeed() && existingSecurity.IsInternalFeed())
+                if (!isInternalFeed && existingSecurity.IsInternalFeed())
```

Comment:
awesome, getting rid of those `security.Property` references

Reply:
Yeah! 👍  were are getting there